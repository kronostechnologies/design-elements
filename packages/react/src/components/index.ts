export * from './accordion';
export * from './avatar';
export * from './badge';
export * from './bento-menu-button';
export * from './breadcrumb';
export * from './buttons';
export * from './card';
export * from './card-link';
export * from './carousel';
export * from './checkbox';
export * from './checkbox-group';
export * from './chooser-button';
export * from './chooser-button-group';
export * from './combobox';
export * from './date-picker';
export * from './design-system';
export * from './device-context-provider';
export * from './disclosure';
export * from './dropdown-list';
export * from './dropdown-menu';
export * from './dropdown-menu-button';
export * from './dropdown-navigation';
export * from './error-summary';
export * from './external-link';
export * from './feedbacks';
export * from './field-container';
export * from './fieldset';
export * from './global-banner';
export * from './global-header';
export * from './global-navigation';
export * from './heading';
export * from './icon';
export * from './internationalization-provider';
export * from './label';
export * from './legend';
export * from './link';
export * from './listbox';
export * from './lozenge';
export * from './menu';
export * from './menu-button';
export * from './modal';
export * from './money-input';
export * from './nav-list';
export * from './numeric-input';
export * from './pagination';
export * from './password-creation-input';
export * from './password-input';
export * from './phone-input';
export * from './progress-circle';
export * from './progress-circular';
export * from './progress-indicator';
export * from './progress-tracker';
export * from './radio-button';
export * from './radio-button-group';
export * from './radio-card-group';
export * from './route-link';
export * from './screen-reader-only-text';
export * from './search';
export * from './sectional-banner';
export * from './segmented-control';
export * from './shadow-wrapper';
export * from './side-drawer';
export * from './skip-link';
export * from './slider';
export * from './spinner';
export * from './status';
export * from './stepper-input';
export * from './table';
export * from './tabs';
export * from './tag';
export * from './text-area';
export * from './text-input';
export * from './theme-wrapper';
export * from './toast';
export * from './toggle-switch';
export * from './toggletip';
export * from './tooltip';
export * from './user-profile';
export * from './visually-hidden';
