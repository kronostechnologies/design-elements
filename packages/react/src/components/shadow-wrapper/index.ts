export { ShadowWrapper } from './shadow-wrapper';
export type { ShadowWrapperProps } from './shadow-wrapper';
