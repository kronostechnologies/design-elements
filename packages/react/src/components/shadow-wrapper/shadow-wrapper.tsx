import React, { createContext, ReactElement, ReactNode, RefObject, useContext, useRef } from 'react';
import root from 'react-shadow/styled-components';
import { mainCss } from '../../styles';

interface ShadowWrapperProps {
    children: ReactNode;
    /**
     * Sets host element tag
     * @default div
     */
    tagName?: string;
}

type ShadowWrapperRef = RefObject<HTMLElement> | null;

const ShadowWrapperRefContext = createContext<ShadowWrapperRef>(null);

export function useShadowWrapperRefContext(): ShadowWrapperRef {
    return useContext(ShadowWrapperRefContext);
}

export function ShadowWrapper({ children, tagName = 'div' }: ShadowWrapperProps): ReactElement {
    const WrapperTag = root[tagName];
    const wrapperRef: ShadowWrapperRef = useRef(null);

    return (
        <WrapperTag ref={wrapperRef}>
            <ShadowWrapperRefContext.Provider value={wrapperRef}>
                <style>{mainCss}</style>
                {children}
            </ShadowWrapperRefContext.Provider>
        </WrapperTag>
    );
}
