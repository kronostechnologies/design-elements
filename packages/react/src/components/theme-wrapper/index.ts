export { ThemeWrapper } from './theme-wrapper';
export type { ThemeWrapperProps } from './theme-wrapper';
