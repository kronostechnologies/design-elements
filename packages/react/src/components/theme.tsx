import React, { ReactElement } from 'react';
import { ThemeProvider } from 'styled-components';

// @ts-ignore
import equisoftTheme from '../themes/equisoft';
// @ts-ignore
export function Theme({ children }): ReactElement {
    return (
        <ThemeProvider theme={equisoftTheme}>
            {children}
        </ThemeProvider>
    );
}
