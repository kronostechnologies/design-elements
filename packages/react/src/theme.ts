export { equisoftTheme } from './themes/equisoft';
export { ResolvedTheme } from './themes/tokens/theme';
