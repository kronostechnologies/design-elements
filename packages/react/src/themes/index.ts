export * from './equisoft';
export * from './default-theme';
export * from './wario-theme';
export * from './tokens';
export * from './build-theme';
