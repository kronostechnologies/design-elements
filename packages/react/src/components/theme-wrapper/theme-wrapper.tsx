import React, { ReactElement } from 'react';
import { ThemeProvider } from 'styled-components';

import {Â equisoftTheme } from '../../themes/equisoft';

export interface Theme {
    main: {
        'primary-1.1': string,
        'primary-1.2': string,
        'primary-1.3': string,
        'primary-2': string,
        'primary-3': string,
        'secondary-4.1': string,
        'secondary-4.2': string,
        'secondary-4.3': string,
    };
    greys: {
        'white': string,
        'colored-white': string,
        'light-grey': string,
        'grey': string,
        'mid-grey': string,
        'dark-grey': string,
        'black': string,
    };
    notifications: {
        'success-1.1': string,
        'success-1.2': string,
        'success-1.3': string,
        'error-2.1': string,
        'error-2.2': string,
        'alert-3.1': string,
        'alert-3.2': string,
        'alert-3.3': string,
    };
}

interface ThemeWrapperProps {
    children: ReactElement;
    theme?: Theme;
}

export function ThemeWrapper({ children, theme }: ThemeWrapperProps): ReactElement {
    let selectedTheme = theme;
    if (selectedTheme) {
        if (Object.entries(selectedTheme).length === 0 && selectedTheme.constructor === Object) {
            selectedTheme = equisoftTheme;
        }
    } else {
        selectedTheme = equisoftTheme;
    }
    return (
        <ThemeProvider theme={selectedTheme}>
            {children}
        </ThemeProvider>
    );
}
