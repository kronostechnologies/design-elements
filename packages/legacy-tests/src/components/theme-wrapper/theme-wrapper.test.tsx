import { render, shallow } from 'enzyme';
import { buildTheme, Button } from '~/components/..';
import { ThemeCustomization } from '~/components/../themes/theme';
import { ShadowWrapper } from '~/components/shadow-wrapper/shadow-wrapper';
import { ThemeWrapper } from '~/components/theme-wrapper/theme-wrapper';

describe('Theme Wrapper', () => {
    it('Returns component with default theme', () => {
        const tree = render(
            <ThemeWrapper>
                <Button buttonType="primary" />
            </ThemeWrapper>,
        );

        expect(tree).toMatchSnapshot();
    });

    it('Returns component with custom theme', () => {
        const themeCustomization: ThemeCustomization = {
            ref: {
                'color-brand-50': '#00874E',
            },
            component: {
                'button-primary-background-color': 'color-brand-50',
            },
        };

        const builtTheme = buildTheme(themeCustomization);

        const tree1 = render(
            <ThemeWrapper theme={builtTheme}>
                <Button buttonType="primary" />
            </ThemeWrapper>,
        );

        expect(tree1).toMatchSnapshot();
    });

    it('should not use ShadowWrapper by default', () => {
        const wrapper = shallow(<ThemeWrapper />);

        expect(wrapper.find(ShadowWrapper)).toHaveLength(0);
    });

    it('should use ShadowWrapper when styles are isolated', () => {
        const wrapper = shallow(<ThemeWrapper isolateStyles />);

        expect(wrapper.find(ShadowWrapper)).toHaveLength(1);
    });
});
