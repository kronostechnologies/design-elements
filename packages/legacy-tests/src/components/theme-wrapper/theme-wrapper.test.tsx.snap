// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Theme Wrapper Returns component with custom theme 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background: inherit;
  border: 1px solid;
  border-radius: 1.5rem;
  box-sizing: border-box;
  color: inherit;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-family: inherit;
  font-size: 0.75rem;
  font-weight: var(--font-bold);
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-letter-spacing: 0.025rem;
  -moz-letter-spacing: 0.025rem;
  -ms-letter-spacing: 0.025rem;
  letter-spacing: 0.025rem;
  line-height: 1rem;
  min-height: var(--size-2x);
  min-width: 2rem;
  outline: none;
  padding: 0 var(--spacing-2x);
  text-transform: uppercase;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c0 {
  outline: 2px solid transparent;
  outline-offset: -2px;
}

.c0:focus-visible {
  box-shadow: 0 0 0 2px #00874E;
  outline: 2px solid #84C6EA;
  outline-offset: -2px;
}

.c0 > svg {
  height: var(--size-1x);
  width: var(--size-1x);
}

.c1 {
  background-color: #00874E;
  border-color: #00874E;
  color: #FFFFFF;
}

.c1 {
  outline: 2px solid transparent;
  outline-offset: -2px;
}

.c1:focus-visible {
  box-shadow: 0 0 0 2px #00874E;
  outline: 2px solid #84C6EA;
  outline-offset: -2px;
}

.c1:hover,
.c1[aria-expanded='true'] {
  background-color: #003A5A;
  border-color: #003A5A;
  color: #FFFFFF;
}

.c1[aria-disabled='true'] {
  background-color: #84C6EA;
  border-color: #84C6EA;
  color: #FFFFFF;
  cursor: not-allowed;
  pointer-events: none;
}

<button
  class="c0 c1"
  type="button"
/>
`;

exports[`Theme Wrapper Returns component with default theme 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background: inherit;
  border: 1px solid;
  border-radius: 1.5rem;
  box-sizing: border-box;
  color: inherit;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-family: inherit;
  font-size: 0.75rem;
  font-weight: var(--font-bold);
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-letter-spacing: 0.025rem;
  -moz-letter-spacing: 0.025rem;
  -ms-letter-spacing: 0.025rem;
  letter-spacing: 0.025rem;
  line-height: 1rem;
  min-height: var(--size-2x);
  min-width: 2rem;
  outline: none;
  padding: 0 var(--spacing-2x);
  text-transform: uppercase;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c0 {
  outline: 2px solid transparent;
  outline-offset: -2px;
}

.c0:focus-visible {
  box-shadow: 0 0 0 2px #006296;
  outline: 2px solid #84C6EA;
  outline-offset: -2px;
}

.c0 > svg {
  height: var(--size-1x);
  width: var(--size-1x);
}

.c1 {
  background-color: #006296;
  border-color: #006296;
  color: #FFFFFF;
}

.c1 {
  outline: 2px solid transparent;
  outline-offset: -2px;
}

.c1:focus-visible {
  box-shadow: 0 0 0 2px #006296;
  outline: 2px solid #84C6EA;
  outline-offset: -2px;
}

.c1:hover,
.c1[aria-expanded='true'] {
  background-color: #003A5A;
  border-color: #003A5A;
  color: #FFFFFF;
}

.c1[aria-disabled='true'] {
  background-color: #84C6EA;
  border-color: #84C6EA;
  color: #FFFFFF;
  cursor: not-allowed;
  pointer-events: none;
}

<button
  class="c0 c1"
  type="button"
/>
`;
